kde-frameworks/karchive/karchive6-6.2.0.recipe

```diff
@@ -0,0 +1,88 @@
+SUMMARY="Reading, creating, and manipulating file archives"
+DESCRIPTION="KArchive provides classes for easy reading, creation and \
+manipulation of /"archive/" formats like ZIP and TAR.
+
+It also provides transparent compression and decompression of data, like the \
+GZip format, via a subclass of QIODevice."
+HOMEPAGE="https://invent.kde.org/frameworks/karchive"
+COPYRIGHT="2010-2024 KDE Organisation"
+LICENSE="GNU LGPL v2"
+REVISION="1"
+SOURCE_URI="https://download.kde.org/stable/frameworks/${portVersion%.*}/karchive-${portVersion}.tar.xz"
+CHECKSUM_SHA256="1435e09e64bf4cf27ceebfc76582e44db9d37b1453e55aeee09778bccfd9a8b2"
+SOURCE_DIR="karchive-$portVersion"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	karchive6$secondaryArchSuffix = $portVersion
+	lib:libKF6archive$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	lib:libQt6Core$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libzstd$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	karchive6${secondaryArchSuffix}_devel = $portVersion
+	devel:libKF6archive$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	karchive6$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	extra_cmake_modules$secondaryArchSuffix >= $portVersion
+	qt6_tools${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:liblzma$secondaryArchSuffix
+	devel:libQt6Core$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libzstd$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:dot
+	cmd:doxygen >= 1.9
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	cmake -Bbuild -S. -DCMAKE_BUILD_TYPE=Release \
+		$cmakeDirArgs \
+		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
+		-DBUILD_QCH=ON \
+		-DBUILD_TESTING=OFF
+	make -Cbuild $jobArgs
+}
+
+INSTALL()
+{
+	make -Cbuild install
+
+	prepareInstalledDevelLib \
+		libKF6Archive
+
+	packageEntries devel \
+		$developDir \
+		$dataDir/doc \
```

Comment:
Shouldn't these files go somewhere in `$documentationDir` (or maybe `$developDocDir`). `$dataDir/doc` looks like a "half-Linuxism" for me (where `share/doc` is the equivalent of `$documentationDir`).

I suspect that the CMake doesn't use `${CMAKE_INSTALL_DOCDIR}` but something like `${CMAKE_INSTALL_DATADIR}/doc`. Could this be patched?

Reply:
I left it as is yes, could be patched, but then we would have to do this for all frameworks that use this which (imho) would lead to more noise at the moment?
I'm not moving them "at this moment" while pushing them, can have a look later though. :)

Reply:
As it looks like it's using datarootdir for the paths (qtqchdir), fidling a bit with it with 6.3.0 to sort this out.
Using: `-DKDE_INSTALL_QTQCHDIR=$documentationDir/packages/KF6` at the moment as it concerns the KF6 frameworks, Qt6 Assistant doesn't pick it up (even if I use the qt6-base directory in there), so you have to manually add it in there to use it, at least I think it would be easier to have them all in one place then in: -DKDE_INSTALL_QTQCHDIR=$documentationDir/packages/ki18n (and the likes)?

Reply:
I'm no expert in this and haven't tested anything myself. So I trust your judgement here fully :)

Reply:
Thanks :) Just hope no one steps up after x version further. :)

Reply:
From Arch: usr/share/doc/qt6/KF6I18n.qch ...